### DISCUSSION TITLE
Criando Apps Web com Angular 8 e Ionic 4

### DISCUSSION THREAD

**Discussion by lxxzh** [balta-io/forum#256] [Tira Dúvidas] [answered]
Boa tarde @andrebaltieri, tudo bom? Estou realizando o curso de Criando Apps Web com Angular 8 e Ionic 4, e finalizando a aula "RESET PASSWORD" notei que meus arquivos **login.page.ts** e **security.util.ts** estão com erros conforme as imagens abaixo. Até copiei as linhas de código do próprio repositório do curso aqui no GITHUB, já procurei algo aqui no fórum e sem sucesso. Poderia me dar um apoio?

<img width="954" alt="Screenshot_2" src="https://user-images.githubusercontent.com/106619427/217643876-cdff0cab-5b64-4371-84b2-97d0ab17ae08.png">
<img width="958" alt="Screenshot_3" src="https://user-images.githubusercontent.com/106619427/217643894-03e4b0cc-69a8-413b-a89b-4fa139776419.png">
<img width="958" alt="Screenshot_4" src="https://user-images.githubusercontent.com/106619427/217643910-8b17d9e7-e883-48da-bf48-3769f0a6831a.png">

**As versões do Angular, Node, IONIC estão na mesma versão da do curso.**

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @lxxzh, como vai?

Em relação ao primeiro e terceiro erro, são referêntes ao nullable types nas novas versões do TypeScript.
É só mudar o retorno para nullable object que resolve:
```ts
public static get(): UserModel? {...

(res: UserModel?) {...
```

Sobre o segundo erro, acredito que a versão do Toast usada no seu projeto, está diferente da do curso.

**Reply by lxxzh** [reply]
Boa noite Balta, em relação ao ToastController, instalei o pacote na mesma versão do curso e sumiu o erro.

Mas em relação ao erro 1 e 3 fiz da forma que mencionou acima e continua o erro e gerou mais erros. Eu fiz da forma correta?

<img width="889" alt="Screenshot_5" src="https://user-images.githubusercontent.com/106619427/217992560-5089006b-af3b-4b14-bd2f-0afff0fbc33b.png">

<img width="568" alt="Screenshot_6" src="https://user-images.githubusercontent.com/106619427/217992583-8f6a0b63-4160-4256-8e64-584b06ed1615.png">

Obs: Mesmo voltando o TS para a mesma versão do curso `"typescript": "~3.4.3",` ainda continuou o erro.

**Reply by andrebaltieri** [reply]
Tenta voltar o TypeScript pra mesma versão do curso e não usar os `?`. O null check foi implementado apenas nas novas versões do TypeScript.

**Reply by lxxzh** [reply]
Bom dia Balta, voltei o TS para a mesma versão e removi os `?`, os erros no código não sumiram e agora está dando este erro quando vou iniciar o IONIC SERVE:


<img width="959" alt="Screenshot_8" src="https://user-images.githubusercontent.com/106619427/218108177-bbddf25d-2e39-4d5e-8cc8-b932ec768de3.png">

**Olha como está meu PACKAGE.JSON:**
<img width="773" alt="Screenshot_9" src="https://user-images.githubusercontent.com/106619427/218109152-84d3c92b-bd1c-4fa6-a947-bfb228669a05.png">
<img width="772" alt="Screenshot_10" src="https://user-images.githubusercontent.com/106619427/218109166-a841a4af-98a9-414f-ba7d-229fbd6a3c00.png">

**Reply by andrebaltieri** [reply]
Bom dia, @lxxzh, acredito que o Ionic CLI esteja em versão diferente.

**Reply by lxxzh** [reply]
Bom dia @andrebaltieri, qual versão está a do curso? Não é a 5.2.4? 
Estou usando essa:
<img width="372" alt="image" src="https://user-images.githubusercontent.com/106619427/218461586-bddeebf9-2689-4c7e-b280-88d4b6404010.png">

**Reply by andrebaltieri** [reply]
@lxxzh não! Acredito que é a 4.x! Não me recordo de ter gravado algo com Ionic 5.

**Reply by lxxzh** [reply]
Boa tarde @andrebaltieri, é que eu segui o passo a passo do curso, conforme a aula "FERRAMENTAS" você instala a versão 5.2.4

<img width="921" alt="image" src="https://user-images.githubusercontent.com/106619427/218824092-96ba4324-e648-4e61-91cd-05919598b375.png">

**Reply by lxxzh** [reply]
E mesmo voltando para a versão IONIC 4.0, dá outros erros:

<img width="957" alt="Screenshot_11" src="https://user-images.githubusercontent.com/106619427/218824318-9ca70233-5404-459b-bfe3-12e312126412.png">

**Reply by andrebaltieri** [reply]
Boa tarde, @lxxzh, como vai?

Tenta o seguinte... 
- Mantém os pacotes e CLIs nas mesmas versões que mostrado no curso
- Na raiz do seu projeto, apaga a pasta `node_modules`
- No terminal, executa o comando `npm cache clean` para limpar o cache
- Executa novamente o comando `npm install` na raiz do projeto para instalar os pacotes necessários
- Executa a aplicação.

**Reply by lxxzh** [reply]
Boa tarde @andrebaltieri, tudo bom?
Fiz todo o processo que você orientou e mesmo assim ainda exibe o mesmo erro.
<img width="953" alt="Screenshot_13" src="https://user-images.githubusercontent.com/106619427/220199585-5be229cc-dde5-4526-b334-b69b943a62cc.png">

**Reply by andrebaltieri** [reply]
@lxxzh bom dia! Se você baixar o código do curso e tentar rodar dá o mesmo erro?

**Reply by lxxzh** [reply]
Boa tarde @andrebaltieri , baixei o código do curso e renderizou. Meu problema deve ser algum erro de pacote. Mas enfim.

<img width="955" alt="Screenshot_14" src="https://user-images.githubusercontent.com/106619427/223194483-61fef3db-91f5-410e-9644-4b4b8ce8a754.png">

Baixei o código do curso e vou acompanhar as aulas, mas ao tentar logar na aplicação não estou conseguindo. Sabe me dizer o porque isto ocorre? O Mockoon está rodando.

**Reply by andrebaltieri** [reply]
@lxxzh show! Seria legal debugar o código pra ver o lance do usuário e senha... não me recordo destas informações... talvez até a senha esteja **hard coded** no código.

**Reply by lxxzh** [reply]
Vou fazer isso, muito obrigado!
